src/client/app/components/ChartLinkComponent.tsx

```diff
@@ -5,21 +5,36 @@
 import * as React from 'react';
 import { FormattedMessage } from 'react-intl';
 import { Button } from 'reactstrap';
+import {ChartTypes} from '../types/redux/graph';
+import {getRangeSliderInterval} from './DashboardComponent';
 
 interface ChartLinkProps {
 	linkText: string;
+	chartType: ChartTypes;
 }
 
 interface ChartLinkState {
 	showLink: boolean;
+	showSliderRange: boolean;
+	sliderRange: string;
+	showOptionalLink: boolean;
+	optionalLink: string;
+	hideOptionsInLinkedPage: boolean;
 }
 
 export default class ChartLinkComponent extends React.Component<ChartLinkProps, ChartLinkState> {
 	constructor(props: ChartLinkProps) {
 		super(props);
 		this.toggleLink = this.toggleLink.bind(this);
+		this.handleOptionsVisibility = this.handleOptionsVisibility.bind(this);
+		// this.handle7DaysChange = this.handle7DaysChange.bind(this);
```

Comment:
Should this commented out line of code be here?

Reply:
I believe since we now fixed compare chart, this feature is not that relevant anymore because compare chart will be able to do the same work. I will remove this part from code.